src/components/nested/NestedComponent.js

```diff
@@ -42,6 +42,18 @@ export default class NestedComponent extends BaseComponent {
     return this.components;
   }
 
+  getAllComponents() {
+    return this.getComponents().reduce((components, component) => {
```

Comment:
This will cause serious performance issues since you are iterating through ALL components at every level of the form. Imagine there is a form with 3 levels deep of nested components. This will iterate over the 3rd level deep components several times since at every level you are iterating over ALL components.